source/broadband/webpa_notification.c

```diff
@@ -354,6 +378,114 @@ void *FactoryResetCloudSync()
 	return NULL;
 }
 
+void SyncNotifyRetryTask()
+{
+	int err = 0;
+	pthread_t threadId;
+	err = pthread_create(&threadId, NULL, SyncNotifyRetry, NULL);
+	if (err != 0)
+	{
+		WalError("Error creating SyncNotifyRetry thread :[%s]\n", strerror(err));
+	}
+	else
+	{
+		WalInfo("SyncNotifyRetry Thread created Successfully\n");
+		g_syncRetryThreadStarted = 1;
+		g_checkSyncNotifyRetry = 1;
+	}
+}
+
+void *SyncNotifyRetry()
+{
+	pthread_detach(pthread_self());
+	char *dbCMC = NULL;
+	int RetryTime = 420;//7mins
+	struct timespec ts;
+	int  rv;
+	int retry_count = 0;    
+	WalInfo("SyncNotifyRetry: max_retry_sleep is 15mins\n");
+
+	while(FOREVER())
+	{
+		clock_gettime(CLOCK_REALTIME, &ts);
+		ts.tv_sec += RetryTime;
+		//wait for backoff delay for retransmission
+		if(g_checkSyncNotifyRetry == 1)
+		{
+			WalInfo("Wait for backoffRetryTime %d sec to check sync notification retry\n", RetryTime);
+		}
+
+		rv = pthread_cond_timedwait(&sync_condition, &sync_mutex, &ts);
+		if (rv == 0)
+		{
+			WalInfo("Received event signal for retrying sync notification.\n");
+			retry_count = 0;
+			continue;
+		} 
+		else if (rv == ETIMEDOUT) 
+		{
+			WalPrint("SyncNotifyRetry thread: Timeout occurred.\n");
+		} 
+		else 
+		{
+			WalError("SyncNotifyRetry thread: Error occurred: %d\n", rv);	
+			continue;
+		}
+
+		if(g_syncNotifyInProgress == 1)
+		{	
+			WalInfo("PARAM_NOTIFY is in progress\n");
+			retry_count = 0;
+			continue;
+		}
+
+		if(g_checkSyncNotifyRetry || (rv == 0))
```

Comment:
why rv = 0 check is added here? this check will not reach as we are doing continue whenever pthread signal is received and rv is 0